cypress/support/gutenberg/commands/assertions.js

```diff
@@ -238,43 +372,54 @@ export function assertHtmlTag( subject, customSelector = '', expectedValue = '',
 		assertBackend = true,
 		assertFrontend = true,
 		delay = 0,
+		afterFrontendAssert = () => {},
+		afterBackendAssert = () => {},
 	} = options
 
 	cy.wp().then( wp => {
 		cy.publish()
 		cy.wait( delay )
+		const blockPath = getBlockStringPath( wp.data.select( 'core/block-editor' ).getBlocks(), subject.data( 'block' ) )
 
-		const block = wp.data.select( 'core/block-editor' ).getBlock( subject.data( 'block' ) )
-		const saveElement = createElementFromHTMLString( wp.blocks.getBlockContent( block ) )
-		const parsedClassList = Array.from( saveElement.classList ).map( _class => `.${ _class }` ).join( '' )
-
-		cy
-			.get( subject )
-			.then( $block => {
-				// Assert editor HTML tag.
-				if ( assertBackend ) {
-					assert.isTrue(
-						! isEmpty( $block.find( `${ expectedValue }${ customSelector }` ) ),
-						`${ customSelector } must have HTML tag '${ expectedValue }' in Editor'`
-					)
-				}
-
-				// Check if we're asserting the parent element.
-				if ( assertFrontend ) {
-					if ( parsedClassList.match( customSelector ) ) {
-						assert.isTrue(
-							saveElement.tagName === toUpper( expectedValue ),
-							`${ customSelector } must have HTML tag '${ expectedValue }' in Frontend'`
-						)
-					} else {
-						// Otherwise, search the element
+		cy.getBlockAttributes().then( attributes => {
+			const selector = `.${ attributes.className }`
+			cy
+				.get( subject )
+				.then( $block => {
+					// Assert editor classes.
+					if ( assertBackend ) {
 						assert.isTrue(
-							saveElement.querySelector( customSelector ).tagName === toUpper( expectedValue ),
-							`${ customSelector } must have HTML tag '${ expectedValue }' in Frontend'`
+							! isEmpty( $block.find( `${ expectedValue }${ customSelector }` ) ),
+							`${ customSelector } must have HTML tag '${ expectedValue }' in Editor'`
 						)
+						afterBackendAssert()
 					}
-				}
-			} )
+
+					// Assert frontend classes
+					if ( assertFrontend ) {
+						cy.getPostUrls().then( ( { editorUrl, previewUrl } ) => {
+							cy.visit( previewUrl )
+							cy.document().then( doc => {
+								const element = doc.querySelector( selector )
+								if ( element ) {
+									cy.get( selector ).then( $frontendBlock => {
+										assert.isTrue(
+											! isEmpty( $frontendBlock.find( `${ expectedValue }${ customSelector }` ) ),
+											`${ customSelector } must have HTML tag '${ expectedValue }' in Frontend'`
+										)
+									} )
+								}
+							} )
```

Comment:
We already assumed that `customSelector` is not part of the parent element `selector` here. Study the previous implementation. We can further simplify this also by using `Array.from( element.classList )`. Add doc comments.